### ISSUE TITLE
支持 eslint 只引用不声明能力

### ISSUE THREAD

**Issue by cjfff** [dumlj/dumlj-build#3]
因为 import/no-extraneous-dependencies 中 packageDir 只会合并依赖，而真正模块可能并没有声明，这样容易导致发布后项目缺少依赖。
例如：项目中有两模块A,B，AB 均没有相互依赖。当 A 依赖声明 lodash，B 引用了 lodash 并且没有声明依赖。这时通过 packageDir: [folder A, folder B], 这时 eslint 会通过检测，而实际 B 需要告警依赖 lodash 并没有声明。
DEMO: https://stackblitz.com/edit/node-4meg4u?file=.eslintrc.js,package.json,packages%2FA%2Findex.js,packages%2FB%2Findex.js,packages%2FB%2Fpacakge.json

**Comment by cjfff** [comment]
I found the solution [here](https://github.com/import-js/eslint-plugin-import/issues/1174#issuecomment-509965883), there is the brief introduction for it.

It work for the single repository well but when it comes to the monorepo it provider the option which is the [packageDir](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md).

If it includes the all of the repositories in contains in the nomorepo, it'll read the dependencies from another package, that's what we don't need.

So, with this situation we can follow the solution in the issue I posted above.
